src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -1291,13 +1309,30 @@ void ObjectSynchronizer::inflate_helper(oop obj) {
 
 // Can be called from non JavaThreads (e.g., VMThread) for FastHashCode
 // calculations as part of JVM/TI tagging.
-static bool is_lock_owned(Thread* thread, oop obj) {
+static bool is_lock_owned(JavaThread* locking_thread, oop obj) {
```

Comment:
The parameter does not need renaming here, we are asking if some thread is the owner, it is not trying to lock anything. Also you've invalidated the comment by making this take a JavaThread instead of Thread.

Reply:
Yes, fix this comment.  If the locking_thread is null, then the lock isn't owned by anybody.  You can reword the comment to point out that the locking_thread will be null if it's called from non-JavaThreads (eg. VMThread) etc. because the comment is a useful reminder.

Reply:
Add: the parameter should be locking_thread because that's what you're asking about.